src/index.js

```diff
@@ -18,21 +18,45 @@ const TCP = and(IP, base('tcp'))
 const UDP = and(IP, base('udp'))
 const UTP = and(UDP, base('utp'))
 
+const TCP_IPFS = or(
```

Comment:
Yeah, sounds about right. A TCP multiaddr is still valid if it has a IPFS hash in it.